### ISSUE TITLE
meta-k network hyperparameters and training the base model on out-ouf-domain+in-domain data

### ISSUE THREAD

**Issue by ZenBel** [zhengxxn/adaptive-knn-mt#5]
Hello,

thanks for the nice work and the clean implementation, it is very easy to use and understand!

I have a couple of (maybe stupid) questions concerning the methodology:

- In your paper you mention you use the valid set to train the meta-k network for about 5k steps. How was the number of steps chosen (since you are using the valid set as a training set)?
- Would it be possible to train a base model using a combination of general open-source data + IT + Med + Koran + Law and then apply the adaptive-knn-mt technique on IT, Med, Koran, Law using this base model? I can use the in-domain training set to create the datastore and then the valid set to train the meta-k network. This should not cause any data leakage or am I missing something? 

Thanks in advance for the attention,

Z

**Comment by zhengxxn** [comment]
Sorry for the late reply...

1. the validation set is used as train and valid set simultaneously ( so in the training script, we set  --train-subset valid --valid-subset valid )， and we use the checkpoint with the best valid_loss ( you can find it in the saved model folder) to test. Since the meta-k network is simple, the input and output are limited, and just a tiny part of the whole translation network, it does not overfit on the valid set.

2.  Yes, and I think it will have better performance on the in-domain translation.

**Comment by ZenBel** [comment]
Great, thanks for the reply!

I have tried point (2) and indeed I see an increase in performance.

Thanks again,

Z
